doc/adrs/153_selection_dialog_tree_view.adoc

```diff
@@ -44,17 +44,28 @@ type SelectionObject {
 
 === Preferred option
 
+==== View model 
+
 The specifier will choose whether the objects returned by the candidates expression will be displayed as a tree or as a flat list. 
 
 The `SelectionDialogDescription` from the Diagram View model will have two additional boolean attributes:
 
  * `displayedAsTree`: to indicate whether the dialog layout should be a tree.
- * `expendedAtOpening`: to indicate if the tree should be expended by default (that means all tree items are expended at opening)
+ * `expandedAtOpening`: to indicate if the tree should be expanded by default (that means all tree items are expanded at opening)
+
+==== Backend and GraphQL API
+
+In addition, the Selection computed by the `SelectionRender` by the server will now return the containment information.
+
+Since the tree depth is not known by the frontend beforehand, the backend needs to return the whole tree items in a flat way.
 
-In addition, the Selection computed by the `SelectionRender` by the server will now return the containment information:
+The GraphQL subscription query will thus have a fixed depth. The containment information will be owned by a field called `parentId`.
```

Comment:
Are you sure about that? What about an update of the subscription that will modify the depth of the tree?

Reply:
I looked at the TreePropertySection and it uses that kind of data structure to deal with that issue.
I did not understand the first time why we made this choice but when I started to implement this ADR I understood why: we can't build a recursive query without knowing the tree depth.

> Are you sure about that? What about an update of the subscription that will modify the depth of the tree?

I'm not sure to understand: that would suppose to have the depth information updated and sent a new request with the same containment depth with
objects {
   ...
   objects {
  	...
  	objects {
    	etc.

Reply:
Note that it is what we do for diagrams but I suppose that it makes sense to limit the node hierarchy:
(see in diagramFragment.ts)

```
nodes {
    ...nodeFragment
    borderNodes {
      ...nodeFragment
    }
    childNodes {
      ...nodeFragment
      borderNodes {
        ...nodeFragment
      }
      childNodes {
        ...nodeFragment
        borderNodes {
          ...nodeFragment
        }
        childNodes {
          ...nodeFragment
          borderNodes {
            ...nodeFragment
          }
          childNodes {
            ...nodeFragment
            borderNodes {
              ...nodeFragment
            }
            childNodes {
              ...nodeFragment
              borderNodes {
                ...nodeFragment
              }
            }
          }
        }
      }
    }
  }
```